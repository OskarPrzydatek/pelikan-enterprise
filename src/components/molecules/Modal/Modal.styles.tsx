import styled, { css } from 'styled-components';

export const Modal = styled.div`
  width: 100%;
  max-height: 100%;
  display: flex;
  flex-direction: column;
  gap: ${({ theme }) => theme.spaces.xl};
  padding: ${({ theme }) => theme.spaces.l};
  background-color: ${({ theme }) => theme.colors.white};

  @media ${({ theme }) => theme.breakpoints.tablet} {
    width: 60%;
    min-height: 30%;
    max-height: 80%;
  }

  > * {
    text-align: center;
  }
`;

export const ModalCloseWrapper = styled.div`
  display: flex;
  justify-content: flex-end;
  align-items: center;
`;

export const backgroundCSS = css`
  justify-content: center;
  align-items: center;
`;

export const closeIconCSS = css`
  align-self: flex-end;
`;

export const acceptButtonCSS = css`
  align-self: center;
  width: 50%;
`;
