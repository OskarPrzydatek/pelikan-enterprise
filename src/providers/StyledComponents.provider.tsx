import { ThemeProvider } from 'styled-components';

import { GlobalStyleLayout } from '../styles/layouts';
import { theme } from '../styles/theme';

interface IStyledComponentsProvider {
  children: React.ReactNode;
}

export const StyledComponentsProvider: React.FC<IStyledComponentsProvider> = ({
  children,
}: IStyledComponentsProvider) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyleLayout />
      {children}
    </ThemeProvider>
  );
};
