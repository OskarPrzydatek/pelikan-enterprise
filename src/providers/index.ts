export { StyledComponentsProvider } from './StyledComponents.provider';
