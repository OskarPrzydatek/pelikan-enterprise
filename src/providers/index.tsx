import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';

import { GlobalStyleLayout } from '../styles/layouts';
import { theme } from '../styles/theme';

interface IStyledComponentsProvider {
  children: React.ReactNode;
}

interface ITestingPageProvider {
  element: JSX.Element;
}

export const StyledComponentsProvider: React.FC<IStyledComponentsProvider> = ({
  children,
}: IStyledComponentsProvider) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyleLayout />
      {children}
    </ThemeProvider>
  );
};

export const TestingPageProvider: React.FC<ITestingPageProvider> = ({
  element,
}: ITestingPageProvider) => {
  return (
    <StyledComponentsProvider>
      <MemoryRouter initialEntries={['/page1']}>
        <Routes>
          <Route element={element} path="/:page" />
        </Routes>
      </MemoryRouter>
    </StyledComponentsProvider>
  );
};
