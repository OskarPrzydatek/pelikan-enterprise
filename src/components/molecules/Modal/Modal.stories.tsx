import { Meta, StoryObj } from '@storybook/react';

import { Text } from '~/components/atoms';
import { StyledComponentsProvider } from '~/providers';

import { Modal } from './Modal';

const meta: Meta<typeof Modal> = {
  title: 'Molecules/Modal',
  component: Modal,
};

export default meta;
type Story = StoryObj<typeof Modal>;

export const ModalWithLeftSideChildren: Story = {
  render: () => (
    <StyledComponentsProvider>
      <Modal
        acceptLabel="Akceptuj"
        onClickAccept={() => console.log('onClickAccept!')}
        onClickBackground={() => console.log('onClickBackground!')}
        onClickClose={() => console.log('onClickClose!')}
      >
        <Text>Example Text</Text>
      </Modal>
    </StyledComponentsProvider>
  ),
};
