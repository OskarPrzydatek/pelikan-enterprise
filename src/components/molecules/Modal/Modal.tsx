import { Background, Button, Icon } from '~/components/atoms';
import { Icons } from '~/constants';
import { IComponent } from '~/models';

import * as S from './Modal.styles';

interface IModal extends IComponent {
  acceptLabel: string;
  children: React.ReactNode;
  closeIconTestID?: string;
  acceptButtonTestID?: string;
  onClickAccept: () => void;
  onClickClose: () => void;
}

export const Modal: React.FC<IModal> = ({
  acceptLabel,
  children,
  dataTestID,
  acceptButtonTestID,
  closeIconTestID,
  onClickAccept,
  onClickClose,
}: IModal) => {
  return (
    <Background
      css={S.backgroundCSS}
      dataTestID={`${dataTestID!}-background`}
      onClickBackground={onClickClose}
    >
      <S.Modal data-testid={dataTestID}>
        <Icon
          css={S.closeIconCSS}
          dataTestID={closeIconTestID}
          icon={Icons.CLOSE}
          onClick={onClickClose}
        />
        {children}
        <Button
          css={S.acceptButtonCSS}
          dataTestID={acceptButtonTestID}
          onClick={onClickAccept}
        >
          {acceptLabel}
        </Button>
      </S.Modal>
    </Background>
  );
};
