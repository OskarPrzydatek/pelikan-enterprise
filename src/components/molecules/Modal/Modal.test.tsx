import { fireEvent, render, screen } from '@testing-library/react';
import { describe, vi } from 'vitest';

import { StyledComponentsProvider } from '~/providers';

import { Modal } from './Modal';

const mockOnClickAccept = vi.fn();
const mockOnClickClose = vi.fn();

const MockModal = () => (
  <StyledComponentsProvider>
    <Modal
      acceptButtonTestID="molecule-modal-accept"
      acceptLabel="Akceptuj"
      closeIconTestID="molecule-modal-close"
      dataTestID="molecule-modal"
      onClickAccept={mockOnClickAccept}
      onClickBackground={mockOnClickClose}
      onClickClose={mockOnClickClose}
    >
      <p data-testid="modal-content">Example Text</p>
    </Modal>
  </StyledComponentsProvider>
);

describe('Modal', () => {
  test('component snapshot', () => {
    const view = render(<MockModal />);
    expect(view).toMatchSnapshot();
  });

  test('ensure modal is closing correctly', () => {
    render(<MockModal />);
    fireEvent.click(screen.getByTestId('molecule-modal-close'));
    expect(mockOnClickClose).toHaveBeenCalled();
    fireEvent.click(screen.getByTestId('molecule-modal-background'));
    expect(mockOnClickClose).toHaveBeenCalled();
  });

  test('ensure modal accept works correctly', () => {
    render(<MockModal />);
    fireEvent.click(screen.getByTestId('molecule-modal-accept'));
    expect(mockOnClickAccept).toHaveBeenCalled();
  });
});
