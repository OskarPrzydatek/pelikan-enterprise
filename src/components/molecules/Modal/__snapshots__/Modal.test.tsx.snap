// Vitest Snapshot v1

exports[`Modal > component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div />
    <div
      class="sc-dkrFOg ieHbTX"
      data-testid="molecule-modal-background"
    >
      <div
        class="sc-jrcTuL hFCxPh"
        data-testid="molecule-modal"
      >
        <button
          class="sc-jSUZER dkgLl"
          data-testid="molecule-modal-close"
        >
          <svg
            data-testid="close-icon"
            fill="none"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 2.01429L17.9857 0L10 7.98571L2.01429 0L0 2.01429L7.98571 10L0 17.9857L2.01429 20L10 12.0143L17.9857 20L20 17.9857L12.0143 10L20 2.01429Z"
              fill="#013059"
            />
          </svg>
        </button>
        <p
          data-testid="modal-content"
        >
          Example Text
        </p>
        <button
          class="sc-hLBbgP cqISts"
          data-testid="molecule-modal-accept"
        >
          Akceptuj
        </button>
      </div>
    </div>
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
